app/Services/UpdateEmailService.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Brewmap\Services;
+
+use Brewmap\Eloquent\User;
+use Brewmap\Exceptions\User\NewEmailChangingException;
+use Brewmap\Notifications\Mail\EmailChangeNotification;
+use Illuminate\Notifications\AnonymousNotifiable;
+
+class UpdateEmailService
+{
+    protected AnonymousNotifiable $notify;
+
+    public function __construct(AnonymousNotifiable $notify)
```

Comment:
I still don't get it. What the heck is `AnonymousNotifiable`? Why its instance is called `$notify`? You cannot name your variables with verbs, because they are not doing anything.

Reply:
**anonymousNotifiable** or **notification** will be okay ?

Reply:
Since ` User ` model is already using `Notifable` trait
https://github.com/brewmap/brewmap/blob/021d7de73f09bcecf104f397a60b151820ca5600/app/Eloquent/User.php#L41

I would prefer to use notifications like it is explained in Laravel documentation:
https://laravel.com/docs/8.x/notifications#using-the-notifiable-trait

So you could send notifications to given user just by simply:
```php
$user->notify(new EmailChangeNotification());
```
Then you won't have to use  `AnonymousNotifiable` and things like `user_id` or `email` will be accessible in `$notifiable` eg.
```php
protected function verifyRoute(User $notifiable): string
{
       return app()->make(UrlGenerator::class)
            ->temporarySignedRoute("api.email.change", $this->time, [
                "user_id" => $user->id,
                "email" => $user->email
       ]);
}
```

Reply:
but I send notification to new email not already used email by User.

Reply:
I believe that you cando something like this:
```php
$user->email = $newEmail;
$user->notify(new EmailChangeNotification());
```

Reply:
I must send notification with link to update new email and i must have user id or old email.

Reply:
temporarySignedRoute create signature based on new email and user id.